### DISCUSSION TITLE
Why my all Gradients are zeros?

### DISCUSSION THREAD

**Discussion by kameronren** [deepjavalibrary/djl#1063] [Q&A]
I implement a ssd. It can run without Exception, but zhe gradients are allways zeros. below is my code.  djl version is 0.11.0
`try (GradientCollector gc = Engine.getInstance().newGradientCollector()) {
					 
					 NDList results = net.forward(parameterStore, X, true);
					 
					 if(firstTime == 0) {
						 
						 requiresGrad(net.getParameters());
					 }
					 
					 MultiBoxTarget mt = MultiBoxTarget.builder().build();
					 NDList inputs = new NDList(results.get(0));
					 inputs.add(Y.get(0));
					 inputs.add(results.get(1).transpose(0,2,1));
					 
					 NDList outputs =  mt.target(inputs);
					 
					 NDArray l = calc_loss(cls_loss,bbox_loss,new NDList(results.get(1)),new NDList(outputs.get(2)),new NDList(results.get(2)),new NDList(outputs.get(0)),new NDList(outputs.get(1)));
					 
					 
					 gc.backward(l);
					 
					 acc_sum += cls_eval(new NDList(results.get(1)),new NDList(outputs.get(2)));
					 n += outputs.get(2).size();
					 mae_sum += bbox_eval(new NDList(results.get(2)),new NDList(outputs.get(0)),new NDList(outputs.get(1)));
					 m += outputs.get(0).size();
					 
					 
				 }`

**Comment by RafaelaSeguro** [comment]
Hi, I know it has been a while since this question was asked, but I am having a similar problem, my gradients are all zero and I have tried debugging every part of my code and found nothing. Did you ever find an answer?
